framework/arcane-framework/src/main/scala/services/streaming/BackfillGroupingProcessor.scala

```diff
@@ -0,0 +1,52 @@
+package com.sneaksanddata.arcane.framework
+package services.streaming
+
+import models.DataRow
+import models.settings.GroupingSettings
+import services.mssql.MsSqlConnection.{BackFillBatch, DataBatch}
+import services.streaming.base.BatchProcessor
+
+import org.slf4j.{Logger, LoggerFactory}
+import zio.stream.ZPipeline
+import zio.{Chunk, ZIO, ZLayer}
+
+import scala.concurrent.duration.Duration
+import scala.util.{Try, Using}
+
+/**
+ * The batch processor implementation that converts a lazy DataBatch to a Chunk of DataRow.
+ * @param groupingSettings The grouping settings.
+ */
+class BackfillGroupingProcessor(groupingSettings: GroupingSettings) extends BatchProcessor[BackFillBatch, Chunk[DataRow]]:
+
+  /**
+   * Processes the incoming data.
+   *
+   * @return ZPipeline (stream source for the stream graph).
+   */
+  def process: ZPipeline[Any, Throwable, BackFillBatch, Chunk[DataRow]] = ZPipeline
+      .map(this.readBatch)
+      .map(tryDataRow => tryDataRow.get)
```

Comment:
Unsafe option read? Smth like 'collect' mb?

Reply:
It's not option, it's a `Try` monad. I use unsafe read here because in case of an exception I want all streaming graph to fail immediately. Otherwise the graph can produce inconsistent data.